compiler/AST/baseAST.cpp

```diff
@@ -263,6 +263,18 @@ void cleanAst() {
           }
         }
       }
+
+      //
+      // If an internal aggregate type is being deleted, set its global
+      // handle to NULL.
+      //
+      if (!isAlive(at)) {
+        if (at == dtBytes) dtBytes = NULL;
+        if (at == dtString) dtString = NULL;
+        if (at == dtLocale) dtLocale = NULL;
+        if (at == dtOwned) dtOwned = NULL;
+        if (at == dtShared) dtShared = NULL;
```

Comment:
Doesn't this potentially prevent them from getting deleted?  I would expect the check to occur just before deleting the node, and then to set it to NULL just after the deletion

Reply:
The `cleanAst` code traverses all the global vectors (which `dtBytes` and `dtString` are in), and sets those to NULL, but I don't ever think it touches the distinct global handles.

Reply:
Are you saying that the aggregate type stored in these globals will be updated independently because it is in the global vector that is traversed?  That does sound plausible

Reply:
I think that if `dtBytes` is pruned and deleted, the pointer that is deleted and set to NULL is the one contained in the global vector and not the handle `dtBytes`.

Reply:
Relevant code is `compiler/AST/baseAST.cpp:203` the macro `clean_gvec`.

Reply:
Alright, that sounds good to me.  I think this is good to go, assuming it passes a fresh round of testing